apps/system/utils/fscmd.c

```diff
@@ -961,9 +962,100 @@ static int mount_handler(FAR const char *mountpoint, FAR struct statfs *statbuf,
 	return OK;
 }
 
+/****************************************************************************
+ * Name: mountpoint_redirection
+ *
+ * Description:
+ *   mountpoint_redirection : Recursively search mount points from the root.
+ *
+ ****************************************************************************/
+static int mountpoint_redirection(const char *cmd, const char *dirpath, mountpoint_handler_t handler, foreach_mountpoint_t mp_handler)
+{
+	int ret = OK;
+	DIR *dirp = opendir(dirpath);		/* Open the directory */
+
+	if (!dirp) {
+		/* Failed to open the directory */
+		FSCMD_OUTPUT("\t TASH %s: no such directory: %s\n", cmd, dirpath);
+
+		return ERROR;
+	}
+
+	/* Read each directory entry */
+	for (;;) {
+		struct dirent *entryp = readdir(dirp);
+		if (!entryp) {
+			/* Finished with this directory */
+			break;
+		}
+
+		/* Call the handler with this directory entry */
+		if (handler(dirpath, entryp, mp_handler) < 0) {
+			/* The handler reported a problem */
+			ret = ERROR;
+			break;
+		}
+	}
+	closedir(dirp);
+
+	return ret;
+}
+
+static int mountpoint_handler(FAR const char *dirpath, FAR struct dirent *entryp, foreach_mountpoint_t handler)
```

Comment:
please provide function description and input/output argument details.

Reply:
Added.